### ISSUE TITLE
overwriting mandatory_validations in custom Validator fails

### ISSUE THREAD

**Issue by stijnvanhoey** [nicolaiarocci/cerberus#235]
When adapting the mandatory_validations, the validation of the added mandatory validation fails:

```python
from cerberus import Validator

class MyValidator(Validator):
    mandatory_validations = ('empty', 'nullable')

val = MyValidator({'user':{'allowed': ['Myriam', 'Floris']}})
document = {'user': ''}
val.validate(document)
print(val.errors)
```

Produces the following error:

```python
 ...
  File "/cerberus/cerberus/cerberus.py", line 902, in _validate_empty
    self._error(field, errors.EMPTY_NOT_ALLOWED)

  File "/cerberus/cerberus/cerberus.py", line 242, in _error
    constraint = field_definitions[rule]

KeyError: 'empty'
```
I would expect that this kind of adaptation by a custom class should be possible without providing this error?

**Comment by funkyfuture** [comment]
your usecase fails because you declare a rule as mandatory and then not specify that rule in your schema.
a fix should therefore raise an error in such cases while schema validation.
the ’nullable’ rule is the only one that has a default constraint. I guess the reasoning behind is that every field that you declare in a schema is in almost all cases expected to have a value (which is to be validated).
so, imo #235 is the wrong approach, rather a hotfix than a solution.

**Comment by stijnvanhoey** [comment]
In interpreted this differently as I was expecting that `mandatory_validations` are validations that will always be executed with a default value (cfr. nullable is False) even when not added to the schema. However, it can be added to the schema to overwrite the default behavior (nullable as True). `Nullable` is in cerberus mandatory, as in: you do not have to add it to the schema to execute it.

**Comment by nicolaiarocci** [comment]
> your usecase fails because you declare a rule as mandatory and then not specify that rule in your schema. a fix should therefore raise an error in such cases while schema validation.

Agreed. Maybe we could update the API docs with this, so people don't get confused anymore.

**Comment by funkyfuture** [comment]
It also should be simple to let the metaclass add a required rule for mandatory rules into the schema validation schema.

**Comment by funkyfuture** [comment]
@stijnvanhoey i added a fix with 0031381. you may submit a test for this derived from you use-case.
